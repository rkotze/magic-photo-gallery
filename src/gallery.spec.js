import React from 'react';
import Gallery from './gallery';
import renderer from 'react-test-renderer';

describe('Render the gallery', () => {
    let galleryComponent;
    beforeAll(() => {
        galleryComponent = renderer.create(
            <Gallery 
            photoList={
                [
                {
                    photo: '/assets/main01.jpg',
                    title: 'Abstract art ',
                    thumbnail: '/assets/thumb01.jpg'
                },
                {
                    photo: '/assets/main02.jpg',
                    title: 'Riding the wave',
                    thumbnail: '/assets/thumb02.jpg'
                },
                {
                    photo: '/assets/main03.jpg',
                    title: 'Sushi dinner',
                    thumbnail: '/assets/thumb03.jpg'
                },
                {
                    photo: '/assets/main04.jpg',
                    title: 'Cat photo',
                    thumbnail: '/assets/thumb04.jpg'
                }
                ]
            } />
        );
    });

    it('the initial state showing first image selected', () => {
        let gallerySnap = galleryComponent.toJSON();
        expect(gallerySnap).toMatchSnapshot();
    });

    it('go to next image', () => {
        galleryComponent.getInstance().next();
        let gallerySnap = galleryComponent.toJSON();
        expect(gallerySnap).toMatchSnapshot();
    });

    it('go to next image using keyboard', () => {
        galleryComponent.getInstance().arrowKeyPress({
            keyCode: 39
        });
        let gallerySnap = galleryComponent.toJSON();
        expect(gallerySnap).toMatchSnapshot();
    });

    it('go to previous image using keyboard', () => {
        galleryComponent.getInstance().arrowKeyPress({
            keyCode: 37
        });
        let gallerySnap = galleryComponent.toJSON();
        expect(gallerySnap).toMatchSnapshot();
    });

    it('specify a photo to change to', () => {
        galleryComponent.getInstance().changePhoto(3)();
        let gallerySnap = galleryComponent.toJSON();
        expect(gallerySnap).toMatchSnapshot();
    });
});