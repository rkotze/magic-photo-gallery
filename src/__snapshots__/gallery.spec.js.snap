// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the gallery go to next image 1`] = `
<div
  className="gallery"
>
  <div>
    <h3>
      Riding the wave
    </h3>
    <div>
      <img
        alt="Riding the wave"
        className="main-photo"
        src="/assets/main02.jpg"
      />
      <div>
        <button
          className="base-button"
          onClick={[Function]}
        >
          &lt; Previous
        </button>
        <button
          className="base-button"
          onClick={[Function]}
        >
          Random
        </button>
        <button
          className="base-button"
          onClick={[Function]}
        >
          Next &gt;
        </button>
      </div>
    </div>
  </div>
  <div
    className="thumbnail-list"
  >
    <ul>
      <li>
        <img
          alt="Abstract art "
          className={null}
          onClick={[Function]}
          src="/assets/thumb01.jpg"
        />
      </li>
      <li>
        <img
          alt="Riding the wave"
          className="selected"
          onClick={[Function]}
          src="/assets/thumb02.jpg"
        />
      </li>
      <li>
        <img
          alt="Sushi dinner"
          className={null}
          onClick={[Function]}
          src="/assets/thumb03.jpg"
        />
      </li>
      <li>
        <img
          alt="Cat photo"
          className={null}
          onClick={[Function]}
          src="/assets/thumb04.jpg"
        />
      </li>
    </ul>
  </div>
</div>
`;

exports[`Render the gallery go to next image using keyboard 1`] = `
<div
  className="gallery"
>
  <div>
    <h3>
      Sushi dinner
    </h3>
    <div>
      <img
        alt="Sushi dinner"
        className="main-photo"
        src="/assets/main03.jpg"
      />
      <div>
        <button
          className="base-button"
          onClick={[Function]}
        >
          &lt; Previous
        </button>
        <button
          className="base-button"
          onClick={[Function]}
        >
          Random
        </button>
        <button
          className="base-button"
          onClick={[Function]}
        >
          Next &gt;
        </button>
      </div>
    </div>
  </div>
  <div
    className="thumbnail-list"
  >
    <ul>
      <li>
        <img
          alt="Abstract art "
          className={null}
          onClick={[Function]}
          src="/assets/thumb01.jpg"
        />
      </li>
      <li>
        <img
          alt="Riding the wave"
          className={null}
          onClick={[Function]}
          src="/assets/thumb02.jpg"
        />
      </li>
      <li>
        <img
          alt="Sushi dinner"
          className="selected"
          onClick={[Function]}
          src="/assets/thumb03.jpg"
        />
      </li>
      <li>
        <img
          alt="Cat photo"
          className={null}
          onClick={[Function]}
          src="/assets/thumb04.jpg"
        />
      </li>
    </ul>
  </div>
</div>
`;

exports[`Render the gallery go to previous image using keyboard 1`] = `
<div
  className="gallery"
>
  <div>
    <h3>
      Riding the wave
    </h3>
    <div>
      <img
        alt="Riding the wave"
        className="main-photo"
        src="/assets/main02.jpg"
      />
      <div>
        <button
          className="base-button"
          onClick={[Function]}
        >
          &lt; Previous
        </button>
        <button
          className="base-button"
          onClick={[Function]}
        >
          Random
        </button>
        <button
          className="base-button"
          onClick={[Function]}
        >
          Next &gt;
        </button>
      </div>
    </div>
  </div>
  <div
    className="thumbnail-list"
  >
    <ul>
      <li>
        <img
          alt="Abstract art "
          className={null}
          onClick={[Function]}
          src="/assets/thumb01.jpg"
        />
      </li>
      <li>
        <img
          alt="Riding the wave"
          className="selected"
          onClick={[Function]}
          src="/assets/thumb02.jpg"
        />
      </li>
      <li>
        <img
          alt="Sushi dinner"
          className={null}
          onClick={[Function]}
          src="/assets/thumb03.jpg"
        />
      </li>
      <li>
        <img
          alt="Cat photo"
          className={null}
          onClick={[Function]}
          src="/assets/thumb04.jpg"
        />
      </li>
    </ul>
  </div>
</div>
`;

exports[`Render the gallery specify a photo to change to 1`] = `
<div
  className="gallery"
>
  <div>
    <h3>
      Cat photo
    </h3>
    <div>
      <img
        alt="Cat photo"
        className="main-photo"
        src="/assets/main04.jpg"
      />
      <div>
        <button
          className="base-button"
          onClick={[Function]}
        >
          &lt; Previous
        </button>
        <button
          className="base-button"
          onClick={[Function]}
        >
          Random
        </button>
        <button
          className="base-button"
          onClick={[Function]}
        >
          Next &gt;
        </button>
      </div>
    </div>
  </div>
  <div
    className="thumbnail-list"
  >
    <ul>
      <li>
        <img
          alt="Abstract art "
          className={null}
          onClick={[Function]}
          src="/assets/thumb01.jpg"
        />
      </li>
      <li>
        <img
          alt="Riding the wave"
          className={null}
          onClick={[Function]}
          src="/assets/thumb02.jpg"
        />
      </li>
      <li>
        <img
          alt="Sushi dinner"
          className={null}
          onClick={[Function]}
          src="/assets/thumb03.jpg"
        />
      </li>
      <li>
        <img
          alt="Cat photo"
          className="selected"
          onClick={[Function]}
          src="/assets/thumb04.jpg"
        />
      </li>
    </ul>
  </div>
</div>
`;

exports[`Render the gallery the initial state showing first image selected 1`] = `
<div
  className="gallery"
>
  <div>
    <h3>
      Abstract art 
    </h3>
    <div>
      <img
        alt="Abstract art "
        className="main-photo"
        src="/assets/main01.jpg"
      />
      <div>
        <button
          className="base-button"
          onClick={[Function]}
        >
          &lt; Previous
        </button>
        <button
          className="base-button"
          onClick={[Function]}
        >
          Random
        </button>
        <button
          className="base-button"
          onClick={[Function]}
        >
          Next &gt;
        </button>
      </div>
    </div>
  </div>
  <div
    className="thumbnail-list"
  >
    <ul>
      <li>
        <img
          alt="Abstract art "
          className="selected"
          onClick={[Function]}
          src="/assets/thumb01.jpg"
        />
      </li>
      <li>
        <img
          alt="Riding the wave"
          className={null}
          onClick={[Function]}
          src="/assets/thumb02.jpg"
        />
      </li>
      <li>
        <img
          alt="Sushi dinner"
          className={null}
          onClick={[Function]}
          src="/assets/thumb03.jpg"
        />
      </li>
      <li>
        <img
          alt="Cat photo"
          className={null}
          onClick={[Function]}
          src="/assets/thumb04.jpg"
        />
      </li>
    </ul>
  </div>
</div>
`;
